api/base/parsers.py

```diff
@@ -15,16 +16,18 @@ def parse(self, stream, media_type=None, parser_context=None):
         Parses the incoming bytestream as JSON and returns the resulting data
         """
         result = super(JSONAPIParser, self).parse(stream, media_type=media_type, parser_context=parser_context)
+        if not isinstance(result, dict):
+            raise ValidationError("Invalid data. Expected a dictionary but got {}".format(type(result)))
```

Comment:
We should avoid Python-specific terms ("dictionary" and the return value of `type(result)`)--remember that clients may not be using Python.

I think raising the default `ParseError` here will be fine. This will return a 400 with the message `Malformed request.'`.